packages/x-date-pickers-pro/src/MultiInputDateTimeRangeField/MultiInputDateTimeRangeField.tsx

```diff
@@ -115,18 +117,19 @@ const MultiInputDateTimeRangeField = React.forwardRef(function MultiInputDateTim
     className: clsx(className, classes.root),
   });
 
-  const TextField = slots?.textField ?? MuiTextField;
-  const startTextFieldProps: FieldsTextFieldProps = useSlotProps({
+  const TextField =
+    slots?.textField ??
+    (inProps.enableAccessibleFieldDOMStructure ? PickersTextField : MuiTextField);
+  const startTextFieldProps = useSlotProps({
     elementType: TextField,
     externalSlotProps: slotProps?.textField,
-    additionalProps: { autoFocus },
     ownerState: { ...ownerState, position: 'start' },
-  });
-  const endTextFieldProps: FieldsTextFieldProps = useSlotProps({
+  }) as BuiltInFieldTextFieldProps<TEnableAccessibleFieldDOMStructure>;
```

Comment:
I've played around and it seems that it is possible to avoid the casting here.
WDYT about the following diff:
```diff
diff --git a/packages/x-date-pickers-pro/src/MultiInputDateTimeRangeField/MultiInputDateTimeRangeField.tsx b/packages/x-date-pickers-pro/src/MultiInputDateTimeRangeField/MultiInputDateTimeRangeField.tsx
index b4799bab6..241a4b57d 100644
--- a/packages/x-date-pickers-pro/src/MultiInputDateTimeRangeField/MultiInputDateTimeRangeField.tsx
+++ b/packages/x-date-pickers-pro/src/MultiInputDateTimeRangeField/MultiInputDateTimeRangeField.tsx
@@ -15,7 +15,7 @@ import {
   splitFieldInternalAndForwardedProps,
   convertFieldResponseIntoMuiTextFieldProps,
 } from '@mui/x-date-pickers/internals';
-import { PickerValidDate, BuiltInFieldTextFieldProps } from '@mui/x-date-pickers/models';
+import { PickerValidDate } from '@mui/x-date-pickers/models';
 import { PickersTextField } from '@mui/x-date-pickers/PickersTextField';
 import { MultiInputDateTimeRangeFieldProps } from './MultiInputDateTimeRangeField.types';
 import { useMultiInputDateTimeRangeField } from '../internals/hooks/useMultiInputRangeField/useMultiInputDateTimeRangeField';
@@ -120,16 +120,22 @@ const MultiInputDateTimeRangeField = React.forwardRef(function MultiInputDateTim
   const TextField =
     slots?.textField ??
     (inProps.enableAccessibleFieldDOMStructure ? PickersTextField : MuiTextField);
-  const startTextFieldProps = useSlotProps({
+  const startTextFieldProps: MultiInputDateTimeRangeFieldProps<
+    PickerValidDate,
+    TEnableAccessibleFieldDOMStructure
+  > = useSlotProps({
     elementType: TextField,
     externalSlotProps: slotProps?.textField,
     ownerState: { ...ownerState, position: 'start' },
-  }) as BuiltInFieldTextFieldProps<TEnableAccessibleFieldDOMStructure>;
-  const endTextFieldProps = useSlotProps({
+  });
+  const endTextFieldProps: MultiInputDateTimeRangeFieldProps<
+    PickerValidDate,
+    TEnableAccessibleFieldDOMStructure
+  > = useSlotProps({
     elementType: TextField,
     externalSlotProps: slotProps?.textField,
     ownerState: { ...ownerState, position: 'end' },
-  }) as BuiltInFieldTextFieldProps<TEnableAccessibleFieldDOMStructure>;
+  });
 
   const Separator = slots?.separator ?? MultiInputDateTimeRangeFieldSeparator;
   const separatorProps = useSlotProps({
@@ -142,7 +148,7 @@ const MultiInputDateTimeRangeField = React.forwardRef(function MultiInputDateTim
   const fieldResponse = useMultiInputDateTimeRangeField<
     TDate,
     TEnableAccessibleFieldDOMStructure,
-    BuiltInFieldTextFieldProps<TEnableAccessibleFieldDOMStructure>
+    MultiInputDateTimeRangeFieldProps<PickerValidDate, TEnableAccessibleFieldDOMStructure>
   >({
     sharedProps: internalProps,
     startTextFieldProps,
```
If it makes sense, then we ought to apply it to all the fields. 🤔

Reply:
Just, we should not use `PickerValidDate` inside our components because we are supposed to know what the type of the date is and it's `TDate`, `PickerValidDate` is the range of dates our components support.

Reply:
I will apply the following diff:

```diff
diff --git a/packages/x-date-pickers-pro/src/MultiInputDateTimeRangeField/MultiInputDateTimeRangeField.tsx b/packages/x-date-pickers-pro/src/MultiInputDateTimeRangeField/MultiInputDateTimeRangeField.tsx
index b4799bab6..28ace8cfa 100644
--- a/packages/x-date-pickers-pro/src/MultiInputDateTimeRangeField/MultiInputDateTimeRangeField.tsx
+++ b/packages/x-date-pickers-pro/src/MultiInputDateTimeRangeField/MultiInputDateTimeRangeField.tsx
@@ -15,12 +15,15 @@ import {
   splitFieldInternalAndForwardedProps,
   convertFieldResponseIntoMuiTextFieldProps,
 } from '@mui/x-date-pickers/internals';
-import { PickerValidDate, BuiltInFieldTextFieldProps } from '@mui/x-date-pickers/models';
:...skipping...
diff --git a/packages/x-date-pickers-pro/src/MultiInputDateTimeRangeField/MultiInputDateTimeRangeField.tsx b/packages/x-date-pickers-pro/src/MultiInputDateTimeRangeField/MultiInputDateTimeRangeField.tsx
index b4799bab6..28ace8cfa 100644
--- a/packages/x-date-pickers-pro/src/MultiInputDateTimeRangeField/MultiInputDateTimeRangeField.tsx
+++ b/packages/x-date-pickers-pro/src/MultiInputDateTimeRangeField/MultiInputDateTimeRangeField.tsx
@@ -15,12 +15,15 @@ import {
   splitFieldInternalAndForwardedProps,
   convertFieldResponseIntoMuiTextFieldProps,
 } from '@mui/x-date-pickers/internals';
-import { PickerValidDate, BuiltInFieldTextFieldProps } from '@mui/x-date-pickers/models';
+import { PickerValidDate } from '@mui/x-date-pickers/models';
 import { PickersTextField } from '@mui/x-date-pickers/PickersTextField';
-import { MultiInputDateTimeRangeFieldProps } from './MultiInputDateTimeRangeField.types';
+import {
+  MultiInputDateTimeRangeFieldProps,
+  MultiInputDateTimeRangeFieldSlotProps,
+} from './MultiInputDateTimeRangeField.types';
 import { useMultiInputDateTimeRangeField } from '../internals/hooks/useMultiInputRangeField/useMultiInputDateTimeRangeField';
 import { UseDateTimeRangeFieldProps } from '../internals/models/dateTimeRange';
-import { MultiInputRangeFieldClasses } from '../models';
+import { MultiInputRangeFieldClasses, RangePosition } from '../models';
 
 export const multiInputDateTimeRangeFieldClasses: MultiInputRangeFieldClasses =
   generateUtilityClasses('MuiMultiInputDateTimeRangeField', ['root', 'separator']);
@@ -120,16 +123,30 @@ const MultiInputDateTimeRangeField = React.forwardRef(function MultiInputDateTim
   const TextField =
     slots?.textField ??
     (inProps.enableAccessibleFieldDOMStructure ? PickersTextField : MuiTextField);
-  const startTextFieldProps = useSlotProps({
+  const startTextFieldProps = useSlotProps<
+    typeof TextField,
+    MultiInputDateTimeRangeFieldSlotProps<TDate, TEnableAccessibleFieldDOMStructure>['textField'],
+    {},
+    MultiInputDateTimeRangeFieldProps<TDate, TEnableAccessibleFieldDOMStructure> & {
+      position: RangePosition;
+    }
+  >({
     elementType: TextField,
     externalSlotProps: slotProps?.textField,
     ownerState: { ...ownerState, position: 'start' },
-  }) as BuiltInFieldTextFieldProps<TEnableAccessibleFieldDOMStructure>;
-  const endTextFieldProps = useSlotProps({
+  });
+  const endTextFieldProps = useSlotProps<
+    typeof TextField,
+    MultiInputDateTimeRangeFieldSlotProps<TDate, TEnableAccessibleFieldDOMStructure>['textField'],
+    {},
+    MultiInputDateTimeRangeFieldProps<TDate, TEnableAccessibleFieldDOMStructure> & {
+      position: RangePosition;
+    }
+  >({
     elementType: TextField,
     externalSlotProps: slotProps?.textField,
     ownerState: { ...ownerState, position: 'end' },
-  }) as BuiltInFieldTextFieldProps<TEnableAccessibleFieldDOMStructure>;
+  });
 
   const Separator = slots?.separator ?? MultiInputDateTimeRangeFieldSeparator;
   const separatorProps = useSlotProps({
@@ -142,7 +159,7 @@ const MultiInputDateTimeRangeField = React.forwardRef(function MultiInputDateTim
   const fieldResponse = useMultiInputDateTimeRangeField<
     TDate,
     TEnableAccessibleFieldDOMStructure,
-    BuiltInFieldTextFieldProps<TEnableAccessibleFieldDOMStructure>
+    typeof startTextFieldProps
   >({
     sharedProps: internalProps,
     startTextFieldProps,

```

Reply:
Yes, great point. 👍 
After providing the diff, I've noticed that mistake. 🙈